### PR TITLE
Test class

### PR DESCRIPTION
**Pull request by ezrayst**
For 2D and 3D `point`

### CODE REVIEW

File: tests/point_test.cc [hunk 1]
```diff
@@ -1,15 +1,35 @@
...
-  //! Check for id = 0
-  SECTION("Point id is zero") {
-    // long long id = 0;
-    // auto node = std::make_shared<felib::Point>(id, coords);
-    // REQUIRE(node->id() == 0);
-  }
-}
+//! \brief Check that Point will store right values
+//! \details Write arbritrary index and coordinate and store them
+
```
Comment (kks32) [inline tests/point_test.cc]: Remove white space

File: tests/point_test.cc [hunk 2]
```diff
@@ -1,15 +1,35 @@
...
-TEST_CASE("Point base is checked for 1D case", "[point][2D]") {
-  //! Check for id = 0
-  SECTION("Point id is zero") {
-    // long long id = 0;
-    // auto node = std::make_shared<felib::Point>(id, coords);
-    // REQUIRE(node->id() == 0);
-  }
-}
+//! \brief Check that Point will store right values
+//! \details Write arbritrary index and coordinate and store them
```
Comment (kks32) [inline tests/point_test.cc]: Remove this line.

File: tests/point_test.cc [hunk 3]
```diff
@@ -1,15 +1,35 @@
...
-//! \brief Check node base class for 1D case
-TEST_CASE("Point base is checked for 1D case", "[point][2D]") {
-  //! Check for id = 0
-  SECTION("Point id is zero") {
-    // long long id = 0;
-    // auto node = std::make_shared<felib::Point>(id, coords);
-    // REQUIRE(node->id() == 0);
-  }
-}
+//! \brief Check that Point will store right values
```
Comment (kks32) [inline tests/point_test.cc]: `Check Point class`

File: tests/point_test.cc [hunk 4]
```diff
@@ -1,15 +1,35 @@
...
+//! \details Write arbritrary index and coordinate and store them
+
+TEST_CASE("Point base is checked in 2D and 3D", "[Point][2D][3D]") {
+
+  //! test for 3d
+  const unsigned Tdim_3d = 3;
```
Comment (kks32) [inline tests/point_test.cc]: rename to `DIM`

File: tests/point_test.cc [hunk 5]
```diff
@@ -1,15 +1,35 @@
...
+  unsigned index_3d = 0;
```
Comment (kks32) [inline tests/point_test.cc]: rename to `index`

File: tests/point_test.cc [hunk 6]
```diff
@@ -1,15 +1,35 @@
...
+  const std::array<double, Tdim_3d> coordinate_3d{0, 0, 0};
+  std::unique_ptr<Point<Tdim_3d>> node_3d(
+      new Point<Tdim_3d>(index_3d, coordinate_3d));
+
+  REQUIRE(node_3d->id() == 0);
+  REQUIRE(node_3d->coordinates().at(0) == 0);
```
Comment (kks32) [inline tests/point_test.cc]: This is a double you need a tolerance.

File: tests/point_test.cc [hunk 7]
```diff
@@ -1,15 +1,35 @@
...
+  REQUIRE(node_3d->coordinates().at(1) == 0);
+  REQUIRE(node_3d->coordinates().at(2) == 0);
+
+  //! test for 2d
+  const unsigned Tdim_2d = 2;
```
Comment (kks32) [inline tests/point_test.cc]: Have a separate test function for 2D.

**Review by kks32 — CHANGES_REQUESTED** [review]
You need to check for maximum and minimum possible values and values of zero (3 cases) for 2D and 3D.

File: tests/point_test_2d.cc [hunk 8]
```diff
@@ -0,0 +1,44 @@
+// Point test
```
Comment (kks32) [inline tests/point_test_2d.cc]: You don't need 2 different files, you can have them in the same file.

File: tests/point_test_2d.cc [hunk 9]
```diff
@@ -0,0 +1,44 @@
...
+#include <array>
+#include <cmath>
```
Comment (kks32) [inline tests/point_test_2d.cc]: `cmath` should be separate. You don't need this.

File: tests/point_test_2d.cc [hunk 10]
```diff
@@ -0,0 +1,44 @@
...
+#include <vector>
+
+#include "catch.hpp"
+#include "point.h"
+
+//! \brief Check point class
+TEST_CASE("Point base is checked in 2D", "[Point][2D]") {
+
+  const unsigned DIM = 2;
+  const double tol = pow(10, -12);
```
Comment (kks32) [inline tests/point_test_2d.cc]: be consistent with naming. Use `1E-16`. Don't use `pow`.

File: tests/point_test_2d.cc [hunk 11]
```diff
@@ -0,0 +1,44 @@
...
+  std::vector<std::unique_ptr<Point<DIM>>> vec_points;
+
+  unsigned index = 0;
+  std::array<double, DIM> coordinate{0, 0};
+  vec_points.emplace_back(new Point<DIM>(index, coordinate));
+
+  index = 4294967295;
```
Comment (kks32) [inline tests/point_test_2d.cc]: Use `limits` to define min, max ranges, don't use actual numbers.

**Review by kks32 — CHANGES_REQUESTED** [review]

File: tests/point_test_2d.cc (hunk 8 above)
Comment (ezrayst) [reply tests/point_test_2d.cc]: do you want two `sections`? or just one?

Comment (kks32) [reply tests/point_test_2d.cc]: Yes, 2 sections.

**Comment by kks32** [comment]
refer to the issue number you are trying to fix when creating a PR.

File: tests/point_test.cc [hunk 12]
```diff
@@ -1,15 +1,113 @@
...
 
-//! \brief Check node base class for 1D case
-TEST_CASE("Point base is checked for 1D case", "[point][2D]") {
-  //! Check for id = 0
-  SECTION("Point id is zero") {
-    // long long id = 0;
-    // auto node = std::make_shared<felib::Point>(id, coords);
-    // REQUIRE(node->id() == 0);
+//! \brief Check point class
+TEST_CASE("Point base is checked in 2D and 3D", "[Point][2D][3D]") {
```
Comment (kks32) [inline tests/point_test.cc]: Can we have a test case for 2D separate from 3D?

**Review by kks32 — CHANGES_REQUESTED** [review]
Split test cases for 2D/3D.

File: tests/point_test.cc [hunk 13]
```diff
@@ -1,15 +1,107 @@
...
-  //! Check for id = 0
-  SECTION("Point id is zero") {
-    // long long id = 0;
-    // auto node = std::make_shared<felib::Point>(id, coords);
-    // REQUIRE(node->id() == 0);
+//! \brief Check point class in 2D
+TEST_CASE("Point base is checked in 2D", "[Point][2D]") {
+
+  //! 2D Test of Point Class
+  const double tol = 1.E-12;
```
Comment (kks32) [inline tests/point_test.cc]: Rename to `tolerance`

File: tests/point_test.cc [hunk 14]
```diff
@@ -1,15 +1,107 @@
...
+  const unsigned DIM = 2;
+
+  //! at coordinate zeroes
+  SECTION("2D test at zeros") {
```
Comment (kks32) [inline tests/point_test.cc]: 2D test of what? `Coordinates are checked for zero values`.

File: tests/point_test.cc [hunk 15]
```diff
@@ -1,15 +1,107 @@
...
+    unsigned index = 0;
```
Comment (kks32) [inline tests/point_test.cc]: `const` index and coordinates.

File: tests/point_test.cc [hunk 16]
```diff
@@ -1,15 +1,107 @@
...
+    std::array<double, DIM> coordinate{0, 0};
+    auto node = std::unique_ptr<Point<DIM>>(new Point<DIM>(index, coordinate));
+
+    REQUIRE(node->id() == 0);
+    REQUIRE(node->coordinates().at(0) == Approx(0).epsilon(tol));
+    REQUIRE(node->coordinates().at(1) == Approx(0).epsilon(tol));
+  }
+
+  //! at minimum limit
+  SECTION("2D test at minimum limits") {
```
Comment (kks32) [inline tests/point_test.cc]: `Coordinates are checked for minimum values`.

File: tests/point_test.cc [hunk 17]
```diff
@@ -1,15 +1,107 @@
...
+    unsigned index = std::numeric_limits<unsigned>::min();
```
Comment (kks32) [inline tests/point_test.cc]: `const`

File: tests/point_test.cc [hunk 18]
```diff
@@ -1,15 +1,107 @@
...
+
+    REQUIRE(node->id() == std::numeric_limits<unsigned>::min());
+    REQUIRE(node->coordinates().at(0) ==
+            Approx(std::numeric_limits<double>::min()).epsilon(tol));
+    REQUIRE(node->coordinates().at(1) ==
+            Approx(std::numeric_limits<double>::min()).epsilon(tol));
+  }
+
+  //! at maximum limit
+  SECTION("2D test at zeros") {
```
Comment (kks32) [inline tests/point_test.cc]: `Coordinates are checked for maximum values`.

File: tests/point_test.cc [hunk 19]
```diff
@@ -1,15 +1,107 @@
...
+    REQUIRE(node->coordinates().at(1) ==
+            Approx(std::numeric_limits<double>::max()).epsilon(tol));
   }
 }
+
+//! \brief Check point class in 3D
+TEST_CASE("Point base is checked in 3D", "[Point][3D]") {
+
+  //! 3D Test of Point Class
+  const double tol = 1.E-12;
```
Comment (kks32) [inline tests/point_test.cc]: `tolerance`

File: tests/point_test.cc [hunk 20]
```diff
@@ -1,15 +1,107 @@
...
+  const unsigned DIM = 3;
+
+  //! at coordinate zeroes
+  SECTION("3D test at maximum limits") {
+    unsigned index = 0;
```
Comment (kks32) [inline tests/point_test.cc]: const

File: tests/point_test.cc [hunk 21]
```diff
@@ -1,15 +1,107 @@
...
+
+//! \brief Check point class in 3D
+TEST_CASE("Point base is checked in 3D", "[Point][3D]") {
+
+  //! 3D Test of Point Class
+  const double tol = 1.E-12;
+  const unsigned DIM = 3;
+
+  //! at coordinate zeroes
+  SECTION("3D test at maximum limits") {
```
Comment (kks32) [inline tests/point_test.cc]: Here you are starting with the maximum, why? be consistent with 2D.

File: tests/point_test.cc [hunk 22]
```diff
@@ -1,15 +1,107 @@
...
+    auto node = std::unique_ptr<Point<DIM>>(new Point<DIM>(index, coordinate));
+
+    REQUIRE(node->id() == 0);
+    REQUIRE(node->coordinates().at(0) == Approx(0).epsilon(tol));
+    REQUIRE(node->coordinates().at(1) == Approx(0).epsilon(tol));
+    REQUIRE(node->coordinates().at(2) == Approx(0).epsilon(tol));
+  }
+
+  //! at minimum limit
+  SECTION("3D test at zeros") {
```
Comment (kks32) [inline tests/point_test.cc]: Rename sections similar to 2D.

**Review by kks32 — CHANGES_REQUESTED** [review]

**Review by kks32 — APPROVED** [review]

**Pull request merged**